airflow/api_fastapi/common/types.py

```diff
@@ -16,10 +16,72 @@
 # under the License.
 from __future__ import annotations
 
-from pydantic import AfterValidator, AwareDatetime
+import inspect
+from datetime import timedelta
+
+from pydantic import AfterValidator, AliasGenerator, AwareDatetime, BaseModel, BeforeValidator, ConfigDict
 from typing_extensions import Annotated
 
+from airflow.models.mappedoperator import MappedOperator
+from airflow.models.operator import Operator
+from airflow.serialization.serialized_objects import SerializedBaseOperator
 from airflow.utils import timezone
 
 UtcDateTime = Annotated[AwareDatetime, AfterValidator(lambda d: d.astimezone(timezone.utc))]
 """UTCDateTime is a datetime with timezone information"""
+
+
+def _validate_timedelta_field(td: timedelta | None) -> TimeDelta | None:
+    """Validate the execution_timeout property."""
+    if td is None:
+        return None
+    return TimeDelta(
+        days=td.days,
+        seconds=td.seconds,
+        microseconds=td.microseconds,
+    )
+
+
+class TimeDelta(BaseModel):
+    """TimeDelta can be used to interact with datetime.timedelta objects."""
+
+    object_type: str = "TimeDelta"
+    days: int
+    seconds: int
+    microseconds: int
+
+    model_config = ConfigDict(
+        alias_generator=AliasGenerator(
+            serialization_alias=lambda field_name: {
+                "object_type": "__type",
+            }.get(field_name, field_name),
+        )
+    )
+
+
+TimeDeltaWithValidation = Annotated[TimeDelta, BeforeValidator(_validate_timedelta_field)]
+
+
+def get_class_ref(obj: Operator) -> dict[str, str | None]:
+    """Return the class_ref dict for obj."""
+    is_mapped_or_serialized = isinstance(obj, (MappedOperator, SerializedBaseOperator))
+
+    module_path = None
+    if is_mapped_or_serialized:
+        module_path = obj._task_module
+    else:
+        module_type = inspect.getmodule(obj)
+        module_path = module_type.__name__ if module_type else None
+
+    class_name = None
+    if is_mapped_or_serialized:
+        class_name = obj._task_type
+    elif obj.__class__ is type:
+        class_name = obj.__name__
+    else:
+        class_name = type(obj).__name__
+
+    return {
+        "module_path": module_path,
+        "class_name": class_name,
+    }
```

Comment:
Because we don't get the `__class__` and other properties of `task` in `TaskResponse`, so we need to perform `isinstance` and other checks before building the `TaskResponse` model.

Reply:
On a model validator you should be able to access the task I think.

This will push this code into the serialised and avoid having this on its own somewhere that will most likely not be reused.

Reply:
I did try with a model validator initially, but since the special properties like `__class__` of the `task` object are not available, unfortunately I was unable to get all info required to get `class_ref` initialized.

Please let me know if you've a code snippet that's working to get `__class_` and other properties in the model validator, and I'll try it out. Thanks!